src/main/java/courseManager/Outline.java

```diff
@@ -0,0 +1,98 @@
+package courseManager;
+
+import weightScheme.WeightScheme;
+
+import java.util.ArrayList;
+
+public class Outline {
+    private double runningGrade;
+    private double hypothethicalGrade;
+
+    private double percentageCompleted;
+    private int numberOfAssessmentsCompleted; /*do these need to be their own instance variables?*/
+    private ArrayList<Assessment> assessments;
+
+    public Outline () {
+        this.runningGrade = 0.0f;
+        this.hypothethicalGrade = 0.0f;
+        this.assessments = new ArrayList<>();
+    }
+
+    public void setRunningGrade(double runningGrade) {
+
+        this.runningGrade = runningGrade;
+    }
+
+    public void setHypothethicalGrade(double hypothethicalGrade) {
+
+        this.hypothethicalGrade = hypothethicalGrade;
+    }
+
+    public void addAssessment(Assessment assessment) {
+
+        assessments.add(assessment);
+    }
+
+    public void removeAssessment(Assessment assessment) {
+
+        assessments.remove(assessment);
+    }
+
+    public void editAssessment(int index, String title) {
+        assessments.get(index).setTitle(title);
+    }
+    public void editAssessment(int index, WeightScheme scheme) {
+        assessments.get(index).setWeightScheme(scheme);
+    }
+    public void editAssessment(int index, String title, WeightScheme scheme) {
+        assessments.get(index).setTitle(title);
+        assessments.get(index).setWeightScheme(scheme);
+    }
+
+    public Assessment getAssessmentData(int index) {
+        return assessments.get(index);
+    }
+
+    public double getPercentageCompleted() {
+        double percentageCompleted = 0.0f;
+        for (Assessment assessment : assessments) {
+            percentageCompleted += assessment.getSubmittedWeight();
+        }
+        return percentageCompleted;
+    }
+
+    public int getNumberOfAssessmentsCompleted() {
```

Comment:
I think a better name for this function would be getNumberOfAssessmentInstancesCompleted. If you intended for it to get the actual number of instances, maybe do a check to see if the number of submitted instances is equal to the number of total instances (or even add a function for that in the assessment class)

Reply:
You're right, I renamed it since I intended to get the number of completed instances **across** all assessments